openpype/plugins/publish/integrate_hero_version.py

```diff
@@ -187,35 +191,29 @@ def integrate_instance(
             repre["name"].lower(): repre for repre in old_repres
         }
 
-        if old_version:
-            new_version_id = old_version["_id"]
-        else:
-            new_version_id = ObjectId()
-
-        new_hero_version = {
-            "_id": new_version_id,
-            "version_id": src_version_entity["_id"],
-            "parent": src_version_entity["parent"],
-            "type": "hero_version",
-            "schema": "openpype:hero_version-1.0"
-        }
-        schema.validate(new_hero_version)
+        op_session = OperationsSession()
 
-        # Don't make changes in database until everything is O.K.
-        bulk_writes = []
+        new_hero_version = new_hero_version_doc(
+            src_version_entity["_id"],
+            src_version_entity["parent"]
+        )
```

Comment:
Rather pass the id to the `new_hero_version_doc` then change it afterwards
```suggestion
        entity_id = None
        if old_version:
            entity_id = old_version["_id"]
        new_hero_version = new_hero_version_doc(
            src_version_entity["_id"],
            src_version_entity["parent"],
            entity_id=entity_id
        )
```